tests/core/scenario/test_scenario.py

```diff
@@ -32,6 +32,21 @@
 from taipy.core.task.task import Task, TaskId
 
 
+def test_scenario_equals(scenario):
+    scenario_manager = _ScenarioManagerFactory()._build_manager()
+
+    scenario_id = scenario.id
+    scenario_manager._set(scenario)
+
+    # To test if instance is same type
+    task = Task("task", {}, print, [], [], scenario_id)
+
+    scenario_2 = scenario_manager._get(scenario_id)
+    assert scenario == scenario_2
+    assert scenario != scenario_id
+    assert scenario != task
+
```

Comment:
```suggestion
We should add a test closer to our user's case using optionals.

```suggestion
    scenario = tp.create_scenario(scenario_cfg)

    # __eq__
    selected_scenario: Optional[Scenario] = None
    try:
        selected_scenario == scenario
    except AttributeError as ae:
        print(ae)  # 'NoneType' object has no attribute 'id'

    # in, indirectly calls __eq__
    selected_scenario: Optional[Scenario] = None
    try:
        selected_scenario in tp.get_scenarios()
    except AttributeError as ae:
        print(ae)  # 'NoneType' object has no attribute 'id'
```

What do you think?

Reply:
I think all the cases have been covered (None, same Scenario, different type instance). Not sure if it's needed :/

Reply:
I don't see where we check for None values.